rust/routee-compass-core/src/model/traversal/default/speed_traversal_model.rs

```diff
@@ -32,6 +49,18 @@ impl TraversalModel for SpeedTraversalModel {
         let (_, edge, _) = trajectory;
         let distance = BASE_DISTANCE_UNIT.convert(&edge.distance, &self.engine.distance_unit);
         let speed = get_speed(&self.engine.speed_table, edge.edge_id)?;
+
+        let speed = match self.speed_limit {
+            Some((speed_limit, _speed_unit)) => {
```

Comment:
Since the user can provide a speed unit, we have to first call convert to make sure we are correctly comparing speeds in the same unit, right?

Reply:
Yes, the idea is to convert it in the `new` method to match the units of the speed model such that it only has to be converted once.

Reply:
Oh ok, good point. Mind leaving a comment here about that? I'm on board.

Reply:
Yes, good idea, just added